spec/interactors/send_fresh_sheet_spec.rb

```diff
@@ -1,132 +1,152 @@
-require "spec_helper"
+require 'spec_helper'
 
 describe SendFreshSheet do
 
-  let!(:market) { create(:market, name: "Mad Dog Farm n Fry", delivery_schedules: [create(:delivery_schedule)]) }
-  let(:note) { "B flat" } #lol
-
-  it "sends test emails" do
-    context = SendFreshSheet.perform(
-      market: market,
-      commit: "Send Test",
-      email: "hossnfeffer@example.com",
-      note: note)
-    expect(context.success?).to eq(true)
-    expect(context.notice).to eq("Successfully sent a test to hossnfeffer@example.com")
-
-    mail = ActionMailer::Base.deliveries.shift
-    expect(mail).to be
+  let!(:market) { create(:market, name: 'Mad Dog Farm n Fry', delivery_schedules: [create(:delivery_schedule)]) }
+  let(:note) { 'B flat' } #lol
+  let(:fresh_subscription) do
+    create(:subscription_type,
+           keyword: SubscriptionType::Keywords::FreshSheet,
+           name: 'Test Fresh!')
+  end
 
-    assert_fresh_sheet_sent_to mail, market, "hossnfeffer@example.com", note
+  let!(:subscribed_buyer) do
+    user = create(:user, :buyer)
+    create(:organization, :buyer, users:[user], markets:[market])
+    user.subscribe_to(fresh_subscription)
+    user
   end
 
-  context "sending to subscribers" do
-    let(:fresh_subscription) { create(:subscription_type, keyword: SubscriptionType::Keywords::FreshSheet, name: "Test Fresh!") }
+  let!(:subscribed_supplier) do
+    user = create(:user, :supplier)
+    create(:organization, :seller, users:[user], markets:[market])
+    user.subscribe_to(fresh_subscription)
+    user
+  end
 
-    let(:subscribed_buyer) do
-      user = create(:user, :buyer)
-      create(:organization, :buyer, users:[user], markets:[market])
-      user.subscribe_to(fresh_subscription)
-      user
+  context 'finding valid subscribers' do
+    subject(:interactor) do
+      interactor = SendFreshSheet.new
+      interactor.context[:market] = market
+      interactor
     end
 
-    let(:subscribed_supplier) do
-      user = create(:user, :supplier)
-      create(:organization, :seller, users:[user], markets:[market])
-      user.subscribe_to(fresh_subscription)
-      user
+    context 'when market users have been deactivated from an organization' do
+      before :each do
+        UserOrganization.where(user_id: subscribed_buyer.id).update_all(enabled: false)
+      end
+      it 'should not find deactivated organization users' do
+        target_users = subject.send(:fresh_sheet_subscribers)
```

Comment:
@thermistor - I agree about this being awkward. Normally I'd put this search in a service object, but we don't have service objects in LO (due to a large overlap with interactors). In lieu of keeping it as an internal method, where do you recommend I put it ? Note: I already extracted the search out of the `perform` method so that it could be separately testable.